dotnet/src/Agents/AzureAI/Extensions/AgentToolDefinitionExtensions.cs

```diff
@@ -0,0 +1,190 @@
+        }
+
+        return new OpenApiAnonymousAuthDetails();
+    }
+
+    internal static List<string>? GetVectorStoreIds(this AgentToolDefinition agentToolDefinition)
+    {
+        return agentToolDefinition.GetOption<List<object>>("vector_store_ids")?.Select(id => id.ToString()!).ToList();
+    }
+
+    private static AzureFunctionBinding GetAzureFunctionBinding(this AgentToolDefinition agentToolDefinition, string bindingType)
+    {
+        Verify.NotNull(agentToolDefinition.Options);
+
+        var binding = agentToolDefinition.GetRequiredOption<Dictionary<object, object>>(bindingType);
+        if (!binding.TryGetValue("storage_service_endpoint", out var endpointValue) || endpointValue is not string storageServiceEndpoint)
+        {
+            throw new ArgumentException($"The option key '{bindingType}.storage_service_endpoint' is required.");
+        }
+        if (!binding.TryGetValue("queue_name", out var nameValue) || nameValue is not string queueName)
+        {
+            throw new ArgumentException($"The option key '{bindingType}.queue_name' is required.");
+        }
+
+        return new AzureFunctionBinding(new AzureFunctionStorageQueue(storageServiceEndpoint, queueName));
+    }
+
+    private static FileSearchRankingOptions? GetFileSearchRankingOptions(this AgentToolDefinition agentToolDefinition)
+    {
+        string? ranker = agentToolDefinition.GetOption<string>("ranker");
+        float? scoreThreshold = agentToolDefinition.GetOption<float>("score_threshold");
+
+        if (ranker is not null && scoreThreshold is not null)
+        {
+            return new FileSearchRankingOptions(ranker, (float)scoreThreshold!);
+        }
+
+        return null;
+    }
+
+    private static List<ToolConnection> GetToolConnections(this AgentToolDefinition agentToolDefinition)
+    {
+        Verify.NotNull(agentToolDefinition.Options);
+
+        var toolConnections = agentToolDefinition.GetRequiredOption<List<object>>("tool_connections");
+
+        return toolConnections.Select(connectionId => new ToolConnection(connectionId.ToString())).ToList();
+    }
+
+    private static T GetRequiredOption<T>(this AgentToolDefinition agentToolDefinition, string key)
+    {
+        Verify.NotNull(agentToolDefinition);
+        Verify.NotNull(agentToolDefinition.Options);
+        Verify.NotNull(key);
+
+        if (agentToolDefinition.Options?.TryGetValue(key, out var value) ?? false)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException($"The option key '{key}' must be a non null value.");
+            }
+
+            if (value is T expectedValue)
+            {
+                return expectedValue;
+            }
+            throw new InvalidCastException($"The option key '{key}' value must be of type '{typeof(T)}' but is '{value.GetType()}'.");
+        }
+
+        throw new ArgumentException($"The option key '{key}' is required.");
```

Comment:
Update were applicable (other implementations).

```suggestion
        throw new ArgumentException($"The option key '{key}' was not found.");
```